server/webapp/WEB-INF/vm/auth/login.vm

```diff
@@ -123,6 +123,11 @@
                 </div>
                 #end
                 <div class='clear-both'><!-- Clear floats --></div>
+
+#foreach ($authentication_plugin_id in $authentication_plugin_registry.getAuthenticationPlugins())
```

Comment:
yes, it does makes sense. But with the current login page & plugins showing up on that page (as buttons/links) it removes the requirement for plugin to handle welcome page. If we decide to go that way we will have to redirect to plugin (which should mandatorily handle that) along with a nice looking UI. Currently with OAuth, Single sign-on etc work off of redirects so there is no need to build a UI (for plugin dev.)